src/components/Profile/SettingsContainer/index.tsx

```diff
@@ -0,0 +1,147 @@
+import React, { useContext, useState, useEffect } from 'react';
+import { Auth } from 'aws-amplify';
+import { View, Text } from 'react-native';
+import { TouchableOpacity } from 'react-native-gesture-handler';
+import { useNavigation } from '@react-navigation/native';
+import styles from './styles';
+import { AppContext } from '../../../contexts/AppContext';
+import { UserContext } from '../../../contexts/UserContext';
+
+import SettingsBox from '../SettingsBox';
+
+const SettingsContainer = () => {
+  const navigation = useNavigation();
+  const { setAppState } = useContext(AppContext);
+  const { userState, setUserState } = useContext(UserContext);
+
+  // workaround to parse data that may be undefined - to change?
+  const homes = userState.homes.items;
+  const homesString = JSON.stringify(homes);
+  const homesJSON = JSON.parse(homesString);
+  const { home } = homesJSON[0];
+  const curHomeName = homesJSON[0].home.addressLine1.split(' ')[1];
+
+  const [homeData] = useState(home);
+  const [name, setName] = useState(String(userState.displayName));
+  const [email, setEmail] = useState(String(userState.username));
+  const [password, setPassword] = useState('');
+  const [address, setAddress] = useState(
+    String(homeData.addressLine1) === 'null'
+      ? ''
+      : String(homeData.addressLine1),
+  );
+  const [address2, setAddress2] = useState(
+    String(homeData.addressLine2) === 'null'
+      ? ''
+      : String(homeData.addressLine2),
+  );
+  const [homeName, setHomeName] = useState(curHomeName);
+
+  const signOut = async () => {
+    try {
+      await Auth.signOut();
+      setUserState({
+        _deleted: null,
+        _lastChangedAt: 0,
+        _version: 1,
+        createdAt: '',
+        displayName: '',
+        homes: {
+          items: [],
+          nextToken: null,
+          startedAt: null,
+        },
+        id: '',
+        owner: '',
+        updatedAt: '',
+        username: '',
+      });
```

Comment:
On sign out, every component past the login stage should be unmounted, so you shouldn't need to manually empty the User Context. Then, when a user signs in, these fields should automatically be repopulated (via the onEffect which pulls this data in the first place). Can you try removing this and confirm if behavior stays the same?

Reply:
I tried removing this, but found that when a user signs out and another user signs in, the info on the profile page remains populated by that first user's data. When the user context is emptied on sign out, the profile page updates as expected when switching accounts

Reply:
Yeah, we need to clear the context since userState is a context.